restructuredtext_lint/cli.py

```diff
@@ -12,7 +12,7 @@
     VERSION = version_file.read().strip()
 
 
-def _main(filepaths, format='text', stream=sys.stdout, encoding=None):
+def _main(filepaths, format='text', stream=sys.stdout, encoding=None, level=0, fail=0):
```

Comment:
For me controlling what linting level gets treated as a failure is the key use case (for git pre-commit hooks and TravisCI checks), implemented here as ``--fail``. The reporting level here implemented as ``--level`` (previously as ``--report``) is in comparison a minor cosmetic feature.

If you're saying combine both under a single switch ``--level`` controlling what is reported AND considered a linting failure, then I'm happy. Its a simpler API which ought to cover most use cases.